shared/src/test/scala/urldsl/examples/PathSegmentExamples.scala

```diff
@@ -60,13 +60,13 @@ final class PathSegmentExamples extends AnyFlatSpec with Matchers {
     /** You can also "group" several segments into a more meaningful class than a pair or a triplet. */
     case class Stuff(str: String, j: Int)
 
-    (s1 / s2).as((Stuff.apply _).tupled, Stuff.unapply(_: Stuff).get).matchRawUrl(sampleUrl) should be(
+    (s1 / s2).as((t: (String, Int)) => Stuff(t._1, t._2), (s: Stuff) => s match { case Stuff(str, j) => (str, j) }).matchRawUrl(sampleUrl) should be(
```

Comment:
I couldn't find a more elegant way to satisfy both Scala versions.

`unapply` [changing signature](https://scalacenter.github.io/scala-3-migration-guide/docs/incompatibilities/other-changed-features.html)